### PR TITLE
Fixed method fetchShipmentListByArea.

### PR DESCRIPTION
**Pull request by Nir-betesh**
Fixed method fetchShipmentListByArea.

### CODE REVIEW

File: EKrut-Server/src/ekrut/server/db/OrderDAO.java [hunk 1]
```diff
@@ -260,21 +262,27 @@ public boolean updateOrderStatus(int orderId, OrderStatus status) {
 	
 	
 	// For Almog: Added by Nir, get list of orders in Shipment status.
-	public ArrayList<Order> fetchOrderShipmentList(String area){
-		PreparedStatement ps = con.getPreparedStatement("SELECT orderId FROM orders WHERE type = 'SHIPMENT' AND area = ?");
+	public ArrayList<Order> fetchOrderShipmentListByArea(String area){
+		PreparedStatement ps = con.getPreparedStatement(
+				"SELECT orderId, date, status, type, dueDate, clientAddress, location, username"
+				+ "FROM orders"
+				+ "WHERE type = 'SHIPMENT'");
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/OrderDAO.java:267]: You're missing spaces between 'username' and 'FROM' and between 'orders' and 'WHERE'.

File: EKrut-Server/src/ekrut/server/db/OrderDAO.java [hunk 2]
```diff
@@ -260,21 +262,27 @@ public boolean updateOrderStatus(int orderId, OrderStatus status) {
...
 		ArrayList<Order> orderList = new ArrayList<>();
 		try {
 			ps.setString(1, area);
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/OrderDAO.java]: The new query doesn't have any parameters.

Comment (Nir-betesh) [reply EKrut-Server/src/ekrut/server/db/OrderDAO.java]: Fixed

File: EKrut-Server/src/ekrut/server/db/OrderDAO.java [hunk 3]
```diff
@@ -225,54 +227,62 @@ public ArrayList<Order> fetchOrdersByUsername(String username) {
...
 
-	
-	
 	// For Almog: Added by Nir, get list of orders in Shipment status.
-	public ArrayList<Order> fetchOrderShipmentList(String area){
-		PreparedStatement ps = con.getPreparedStatement("SELECT orderId FROM orders WHERE type = 'SHIPMENT' AND area = ?");
+	public ArrayList<Order> fetchOrderShipmentListByArea(String area) {
+		PreparedStatement ps = con.getPreparedStatement(
+				"SELECT o.orderId, o.date, o.status, o.type, o.dueDate, o.clientAddress, o.location, o.username "
+				+ "FROM orders o, ekrut_machines e "
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/OrderDAO.java]: You're not using ekrut_machines for anything here.

File: EKrut-Server/src/ekrut/server/db/OrderDAO.java [hunk 4]
```diff
@@ -225,54 +227,62 @@ public ArrayList<Order> fetchOrdersByUsername(String username) {
...
-			ps.setString(1, area);
+			// Get all orders that their status is SHIPMENT
 			ResultSet rs = con.executeQuery(ps);
-			while(rs.next()) {
-				Order order = fetchOrderById(rs.getInt(1));
-				if (order.getStatus() == OrderStatus.SUBMITTED)
-					orderList.add(order);
+			while (rs.next()) {
+				// Get customer area in order to match the requested area with manager area.
+				User user = userDAO.fetchUserByUsername(rs.getString(7));
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/OrderDAO.java]: The username is the 8th column in the result, not the 7th.

**Pull request merged**